model/user/db.go

```diff
@@ -488,18 +488,31 @@ func ClearLoginCache(user gimlet.User) error {
 
 // ClearUser clears one user's settings, roles, and login cache.
 func ClearUser(userId string) error {
-	update := bson.M{
+	unsetUpdate := bson.M{
 		"$unset": bson.M{
-			SettingsKey:   1,
+			SettingsKey: bson.M{
+				userSettingsGithubUserKey:    1,
+				userSettingsSlackUsernameKey: 1,
+				userSettingsSlackMemberIdKey: 1,
+			},
 			RolesKey:      1,
 			LoginCacheKey: 1,
-		}}
+		},
+	}
 	query := bson.M{IdKey: userId}
-	if err := UpdateOne(query, update); err != nil {
+	if err := UpdateOne(query, unsetUpdate); err != nil {
 		return errors.Wrap(err, "unsetting user settings")
 	}
-
-	return nil
+	setUpdate := bson.M{
+		"$set": bson.M{
+			SettingsKey: bson.M{
+				UseSpruceOptionsKey: bson.M{
+					SpruceV1Key: 1,
+				},
```

Comment:
1 is only used for unset; for set, you want to use the actual value (true, false). In this case, I'm not actually sure waht you're going for. You can also just do this in one update: it would contain both the set and the unset (example: https://github.com/evergreen-ci/evergreen/blob/9f3350a1017ac4c63faa2480e273af4c7706a294/model/task/db.go#L176)

Reply:
Ah that makes sense, alas. What if we just didn't update this at all? Let them keep their default?

Reply:
Or if we're going to reset this anyway, we can just unset the entire user settings.

Reply:
You can just set this to true, not "true". SpruceV1 is a boolean.